app/query/caller.go

```diff
@@ -102,29 +102,25 @@ func (c *Caller) NewRPCClient(timeInSecondstimeOut time.Duration) *jsonrpc.RPCCl
 			},
 		},
 	})
-	return &client
+	return client
 }
 
 func (c *Caller) GetTimeSpanForJSONRPCMethod(method string) time.Duration {
-	var retVal time.Duration
-	switch method {
-	case "txo_spend":
-		retVal = time.Hour * 3
-		break
-	case "txo_list":
-		retVal = time.Minute * 10
-		break
-	case "transaction_list":
-		retVal = time.Minute * 10
-		break
-	default:
-		retVal = time.Minute * 2
+	timeSpanMap := map[string]time.Duration{
+		"txo_spend":        time.Hour * 3,
+		"txo_list":         time.Minute * 10,
+		"transaction_list": time.Minute * 10,
+	}
+
+	if timeSpan, ok := timeSpanMap[method]; ok {
+		return timeSpan
 	}
-	return retVal
+
+	return time.Minute * 2
 }
 
-func (c *Caller) GetRPCClientForMethod(method string) *jsonrpc.RPCClient {
-	var client *jsonrpc.RPCClient = c.NewRPCClient(c.GetTimeSpanForJSONRPCMethod(method))
+func (c *Caller) GetRPCClientForMethod(method string) jsonrpc.RPCClient {
```

Comment:
There's also no point in keeping `GetRPCClientForMethod` and `GetTimeSpanForJSONRPCMethod`  public since newRPCClient is going to be private. Sorry if I was unclear on that!

Reply:
Thank you for pointing that out. They definitely do not need to be exposed. Fixed